### ISSUE TITLE
Setup unclear for Google's OAuth Plugin

### ISSUE THREAD

**Issue by cgsmith** [osTicket/osTicket-plugins#50]
Please document clear instructions on how to generate a key and password for OAuth.  I just get a blank screen.

**Comment by comexpertise** [comment]
+1

Check your logs, you probably not run make.php file for hydrate dependencies?

(aarobc): After cloning the repo, you have to run:

php make.php hydrate

That will pull down the dependencies. Next, you have to move the auth-oath directory and the lib directory up so they're directly in include/plugins/.

Create OAth credentials in google developer settings and enter them into the plugin. 
For the Authorized Redirect URI enter in yourdomainhere/api/auth/ext

Under the Developer console select "Consent Screen" and enter appropriate information.

When I try to log in, I get tho the consent screen, but when I accept, I'm redirected to:
mydomain/api/auth/ext?state=53c4....

And a message saying "The requested URL /api/auth/ext was not found on this server."

I feel I'm close, but what am I missing?

**Comment by ntozier** [comment]
What version of osT are you running?
Side note: the oauth plugin hasn't really been released (see http://osticket.com/download#addons for available plugins).

**Comment by cgsmith** [comment]
Surprisingly kind of hard to find a version number easily: 1.9.3 is the version.  I thought it might not be out officially... but documentation should be there during beta/alpha.

**Comment by comexpertise** [comment]
Read this for [install oauth plugin](http://www.webpreprod.com/issue/utiliser-protocole-oauth-20-osticket-se-connecter-compte-google-facebook)

**Comment by greezybacon** [comment]
I included some basic notes in the [original pull request](https://github.com/osTicket/core-plugins/pull/24)

**Comment by qlx-fmarines** [comment]
Hello everyone, 
does anyone know how to make this plugin up ? i did all the suggestions from all the ticket and support pages i could find in the past week, so far the plugins were pulled from Git repository, hydrated, moved to the correct place(plugins) along the lib folder and the .phar file, google + Api set up correctly, plugin added into the osticket with the correct parameters,.  it works fine at the point to redirect to the previous screen, this is what i have from my log file
[Sat May 21 18:27:33.185282 2016] [proxy_fcgi:error] [pid 9318:tid 140527571760896] [client 10.0.2.2:56278] AH01071: Got error 'PHP message: PHP Fatal error:  Class 'Au
th2' not found in phar:///opt/bitnami/apache2/htdocs/osticket/include/plugins/auth-oauth.phar/google.php on line 15\n'

**Comment by ntozier** [comment]
error "Class Auth2 not found in Phar."
Looks like your missing a dependency.

ohmy/Auth2

A quick google makes me think that its this:
https://github.com/sudocode/ohmy-auth/blob/master/src/ohmy/Auth2.php

**Comment by qlx-fmarines** [comment]
Yes!, you were right, now is working 50%, Users side works fine but Agents side is not, methods from google.php are not getting what it expects, i'm not php programmer is my sense of what is happening,  here is the error i got.   

[Thu May 26 11:34:02.770801 2016] [proxy_fcgi:error] [pid 9708:tid 140166626752256] [client 10.0.2.2:59932] AH01071: Got error 'PHP message: PHP Warning:  explode() exp
ects parameter 2 to be string, array given in /opt/bitnami/apache2/htdocs/osticket/include/pear/Mail/RFC822.php on line 308\nPHP message: PHP Warning:  explode() expect
s parameter 2 to be string, array given in /opt/bitnami/apache2/htdocs/osticket/include/pear/Mail/RFC822.php on line 246\nPHP message: PHP Warning:  substr() expects pa
rameter 1 to be string, array given in /opt/bitnami/apache2/htdocs/osticket/include/pear/Mail/RFC822.php on line 278\n', referer: https://accounts.google.com/ServiceLog
in?passive=1209600&continue=https://accounts.google.com/o/oauth2/auth?scope%3Dprofile%2Bemail%26response_type%3Dcode%26redirect_uri%3Dhttps://xxx.xxx.com/helpdesk/a
pi/auth/ext%26state%3Dd0fc8a1db7c676d303bd0206757f24bf%26client_id%3D732920948782-rokvmeu6a7bcd73efsvffin9ra7qoqf1.apps.googleusercontent.com%26from_login%3D1%26as%3D48
02d962cd5989fa&oauth=1&sarp=1&scc=1

**Comment by ntozier** [comment]
@greezybacon this one is all you. :)

**Comment by sasiryu** [comment]
php make.php hydrate throws error Line:481 can some one please help

**Comment by anandselvarasu** [comment]
I am also facing same issue on google single sign on..

**Comment by Technowise** [comment]
I was able to successfully install the OAuth plugin for Google login by following [this article](http://www.webpreprod.com/issue/utiliser-protocole-oauth-20-osticket-se-connecter-compte-google-facebook) in french.

I have documented the steps for installing this in my blog post here:
http://www.technowise.in/2017/01/osticket-googlegmail-login-integration.html

**Please note**: You may not be able to get it it to work on NGinx as the callback URL from oAuth depends on URL rewrite specified in .htaccess in api folder. So I would recommend you use Apache server if you intend to install and use this plugin.

**Comment by yatinchawla** [comment]
@Technowise I followed your tutorial but I am facing one issue.
When I got to home page i.e
www.example.com and click on Sign in > Sign in with Google
I am able to login successfully with OAuth (as a client)

But when i go to Admin login i.e
www.example.com/scp
and click on Sign in with Google, On first time click nothing happens, second time, I get error as
"Have your administrator create a local account"

Though I have the Google Email registered in my OSticket as an agent with same email id, still its not able to authorise the agent login via OAuth.

**Comment by vishnus** [comment]
fix for nginx guys:

location /api {
     try_files $uri $uri/ /api/http.php?$args;
}

**Comment by cheasingh** [comment]
> fix for nginx guys:
> 
> location /api {
> try_files $uri $uri/ /api/http.php?$args;
> }

thank @vishnus, this work on nginx
